lib/index.js

```diff
@@ -1 +1,121 @@
-// Your code will go here.
+"use strict";
+
+import md5 from "md5";
+import uuid from "uuid";
+import "isomorphic-fetch";
+
+var localStorage;
+if (typeof (localStorage) == "undefined" || localStorage === null) {
+  var LocalStorage = require('node-localstorage').LocalStorage;
+  localStorage = new LocalStorage('./lib/localStorage');
+}
+ function parseHexString(str) {
+  var result = [];
+  while (str.length >= 2) {
+    result.push(parseInt(str.substring(0, 2), 16));
+    str = str.substring(2, str.length);
+  }
+  return result;
+}
+
+export function getUserIDHash(userID) {
+  var hash = md5(userID);
+  var input = parseHexString(hash);
+  return uuid.v4({random: input});
+};
+
+export function registerUserURL(userID, centralRepositoryURL, headers, userStorageURL){
+
+  var user_record_id = getUserIDHash(userID);
+
+  var key = 'kinto:server-url:' + userID;
+  var cachedURL = localStorage.getItem(key);
+  if(cachedURL != null){
```

Comment:
We probably want to validate that the ``cachedURL`` is the new ``userStorageURL`` that we are passing for.

Maybe add a test to make sure that the ``userStorageURL`` can be updated even if a previous ``cachedURL`` is stored in the ``localStorage``.